src/middleware.ts

```diff
@@ -1,10 +1,47 @@
 import { withAuth } from "next-auth/middleware";
+import { NextResponse } from 'next/server'
+import type { NextRequest } from 'next/server'
 
 export const pagesNextAuthOptions = {
   signIn: "/signin",
 };
 
+// Custom middleware
+export function middleware(req: NextRequest) {
+
+  // Checking `/api` routes
+  if (req.nextUrl.pathname.startsWith("/api")) {
```

Comment:
also if i hit /api/auth/signin for google, won't this check if i have the auth token as well? it might just be worth to not add middleware and do the check only in the api sphere webhook part